sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -727,6 +727,19 @@ class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
     }
   }
 
+  test("cast struct with a timestamp field") {
+    val origSchema = StructType(
+      Seq(StructField( "tsField", TimestampType, false ))
+    )
+    val trgSchema = StructType(
+      // nine out of ten times I'm casting a struct, it's to normalize its fields nullability
+      Seq(StructField( "tsField", TimestampType, true ))
+    )
+    val inp = Literal.create(InternalRow(0L), origSchema)
```

Comment:
This is an internal row,timestamps are represented as longs.
We can probably use a more meaningful date literal